parsel/selector.py

```diff
@@ -44,11 +44,14 @@ def _st(st):
         raise ValueError('Invalid type: %s' % st)
 
 
-def create_root_node(text, parser_cls, base_url=None):
+def create_root_node(text, parser_cls, base_url=None, encoding='utf8'):
     """Create root node for text using given parser class.
     """
-    body = text.strip().replace('\x00', '').encode('utf8') or b'<html/>'
-    parser = parser_cls(recover=True, encoding='utf8')
+    if six.PY3 and isinstance(text, bytes):
+        body = text.replace(b'\x00', b'') or b'<html/>'
+    elif isinstance(text, six.text_type):
+        body = text.strip().replace('\x00', '').encode(encoding) or b'<html/>'
```

Comment:
I’ve just realized that the point of strip would be to allow `or b'<html/>'` to work with a space-only body, so that `' '` gets replaced by `b'<html/>'`. To keep that behavior, I believe we need to keep `.strip()`. Also, because [`.strip()` behaves differently for  `str` and `bytes`](https://stackoverflow.com/a/22231264), we should use the method before encoding.

So, what about this?

```suggestion
    if six.PY2 or isinstance(text, str):
        body = text.strip().encode(encoding)
    else:
        body = text.strip()
    body = body.replace(b'\x00', b'') or b'<html/>'
```

Reply:
Related codelines updated https://github.com/GeorgeA92/parsel/commit/89b15f89ddd783931c39bad40f951ae8cddc221f